### DISCUSSION TITLE
Swarm generic resource

### DISCUSSION THREAD

**Discussion by nie3e** [moby/moby#47253] [Q&A]
Can I somehow check the status of swarm generic resources (NVIDIA-GPU)?
Need to know which devices are in use.

**Comment by neersighted** [comment]
Generic resources are that -- generic markers. They have no actual connection to the underlying resources the Swarm administrator is representing with them.

I suppose you might be asking if you can find out how many containers are scheduled on a given worker matching a given generic resource -- @dperny, do you know if that's possible to answer, and how?

**Reply by nie3e** [reply]
This rarely happens, but I have a strange problem:
there are 6 containers - each of them uses 1 GPU.
Sometimes one of them crashes (suppose the one that is using device numer 2), and after restart it gets device number 0 that is already occupied by another container. I am not sure how to track what happens there. The only workaround I have for now is: scale this service to 0 and then back to 6.
